synapse/rest/client/v2_alpha/auth.py

```diff
@@ -178,6 +221,40 @@ def on_POST(self, request, stagetype):
             request.write(html_bytes)
             finish_request(request)
 
+            defer.returnValue(None)
+        elif stagetype == LoginType.TERMS:
+            if ('session' not in request.args or
+                    len(request.args['session'])) == 0:
+                raise SynapseError(400, "No session supplied")
+
+            session = request.args['session'][0]
+            authdict = {'session': session}
+
+            success = yield self.auth_handler.add_oob_auth(
+                LoginType.TERMS,
+                authdict,
+                self.hs.get_ip_from_request(request)
+            )
+
+            if success:
+                html = SUCCESS_TEMPLATE
+            else:
+                html = TERMS_TEMPLATE % {
+                    'session': session,
+                    'terms_url': "%s/_matrix/consent" % (
```

Comment:
I'm slightly worried about a user seeing an old cached version of the consent form, and then we record them as having agreed to a more recent version. Can we include the version in the URI and in the session?